### PR TITLE
Add ROS 2 Jazzy to CI

### PR DESCRIPTION
**Pull request by lreiher**

### CODE REVIEW

File: .github/workflows/docker-ros.yml [hunk 1]
```diff
@@ -4,27 +4,57 @@ on: push
...
     steps:
       - uses: ika-rwth-aachen/docker-ros@main
         with:
+          platform: amd64,arm64
+          target: dev,run
           image-tag: ros2
           base-image: rwthika/ros2:humble
           command: ros2 launch etsi_its_conversion converter.launch.py
+          enable-industrial-ci: 'true'
+          enable-push-as-latest: 'true'
```
Comment (jpbusch) [inline .github/workflows/docker-ros.yml:31]: Only set `enable-push-as-latest` true for one configuration. I would suggest ros2-jazzy.

File: .github/workflows/docker-ros.yml [hunk 2]
```diff
@@ -4,27 +4,57 @@ on: push
...
+
+  ros2-jazzy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ika-rwth-aachen/docker-ros@main
+        with:
+          platform: amd64,arm64
+          target: dev,run
+          image-tag: ros2-jazzy
+          base-image: ubuntu:24.04
```
Comment (jpbusch) [inline .github/workflows/docker-ros.yml:56]: If the `base-image` doesn't contain the ROS installation, we should define the `ros-distro` arg. Alternatively, we could just use `ros:jazzy`, couldn't we?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -5,31 +5,59 @@ Code Generation:
...
     strategy: depend
   variables:
-    PLATFORM: amd64
+    PLATFORM: amd64,arm64
     TARGET: dev,run
     IMAGE_TAG: ros2
     BASE_IMAGE: rwthika/ros2:humble
     COMMAND: ros2 launch etsi_its_conversion converter.launch.py
     ENABLE_INDUSTRIAL_CI: 'true'
     ENABLE_PUSH_AS_LATEST: 'true'
```
Comment (jpbusch) [inline .gitlab-ci.yml:34]: push jazzy as latest?

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -5,31 +5,59 @@ Code Generation:
...
+ros2-jazzy:
+  trigger:
+    include:
+      - remote: https://raw.githubusercontent.com/ika-rwth-aachen/docker-ros/main/.gitlab-ci/docker-ros.yml
+    strategy: depend
+  variables:
+    PLATFORM: amd64,arm64
+    TARGET: dev,run
+    IMAGE_TAG: ros2-jazzy
+    BASE_IMAGE: ubuntu:24.04
```
Comment (jpbusch) [inline .gitlab-ci.yml:60]: same here

File: .github/workflows/docker-ros.yml (hunk 2 above)
Comment (lreiher) [reply .github/workflows/docker-ros.yml:56]: With existing jazzy images, I encountered this error:

```
root@828513742cf8:/# curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
root@828513742cf8:/# echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | tee /etc/apt/sources.list.d/ros2.list
deb [arch=amd64 signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu noble main
root@828513742cf8:/# 
root@828513742cf8:/# apt update
E: Conflicting values set for option Signed-By regarding source http://packages.ros.org/ros2/ubuntu/ noble: /usr/share/keyrings/ros2-latest-archive-keyring.gpg != /usr/share/keyrings/ros-archive-keyring.gpg
E: The list of sources could not be read.
```

We might need to fix something in docker-ros. Will try to use ubuntu:24.04 for now.

**Comment by lreiher** [comment]
Currently blocked by https://github.com/nobleo/rviz_satellite/issues/123

**Review by jpbusch — APPROVED** [review]

**Pull request merged**